src/tests/functional/shared_test_classes/src/single_op/comparison.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include "ngraph_functions/builders.hpp"
+#include "common_test_utils/ov_tensor_utils.hpp"
+
+namespace ov {
+namespace test {
+using ngraph::helpers::ComparisonTypes;
+using ngraph::helpers::InputLayerType;
+
+std::string ComparisonLayerTest::getTestCaseName(const testing::TestParamInfo<ComparisonTestParams> &obj) {
+    std::vector<InputShape> shapes;
+    ComparisonTypes comparison_op_type;
+    InputLayerType second_input_type;
+    ov::element::Type model_type;
+    std::string device_name;
+    std::map<std::string, std::string> additional_config;
+    std::tie(shapes,
+             comparison_op_type,
+             second_input_type,
+             model_type,
+             device_name,
+             additional_config) = obj.param;
+
+    std::ostringstream result;
+    result << "IS=(";
+    for (size_t i = 0lu; i < shapes.size(); i++) {
+        result << ov::test::utils::partialShape2str({shapes[i].first}) << (i < shapes.size() - 1lu ? "_" : "");
+    }
+    result << ")_TS=";
+    for (size_t i = 0lu; i < shapes.front().second.size(); i++) {
+        result << "{";
+        for (size_t j = 0lu; j < shapes.size(); j++) {
+            result << ov::test::utils::vec2str(shapes[j].second[i]) << (j < shapes.size() - 1lu ? "_" : "");
+        }
+        result << "}_";
+    }
+    result << "comparisonOpType=" << comparison_op_type << "_";
+    result << "secondInputType=" << second_input_type << "_";
+    result << "in_type=" << model_type.get_type_name() << "_";
+    result << "targetDevice=" << device_name;
+    return result.str();
+}
+
+void ComparisonLayerTest::SetUp() {
+    std::vector<InputShape> shapes;
+    InputLayerType second_input_type;
+    std::map<std::string, std::string> additional_config;
+    ov::element::Type model_type;
+    std::tie(shapes,
+             comparison_op_type,
+             second_input_type,
+             model_type,
+             targetDevice,
+             additional_config) = this->GetParam();
+    configuration.insert(additional_config.begin(), additional_config.end());
+    init_input_shapes(shapes);
+
+    ov::ParameterVector inputs {std::make_shared<ov::op::v0::Parameter>(model_type, inputDynamicShapes[0])};
+
+    std::shared_ptr<ov::Node> second_input;
+    if (second_input_type == InputLayerType::PARAMETER) {
+        second_input = std::make_shared<ov::op::v0::Parameter>(model_type, inputDynamicShapes[1]);
+        inputs.push_back(std::dynamic_pointer_cast<ov::op::v0::Parameter>(second_input));
+    } else {
+        ov::Tensor tensor = ov::test::utils::create_and_fill_tensor(model_type, targetStaticShapes.front()[1]);
+        second_input = std::make_shared<ov::op::v0::Constant>(tensor);
+    }
+
+    auto comparisonNode = ngraph::builder::makeComparison(inputs[0], second_input, comparison_op_type);
+    function = std::make_shared<ov::Model>(comparisonNode, inputs, "Comparison");
+}
+
+void ComparisonLayerTest::generate_inputs(const std::vector<ov::Shape>& target_input_static_shapes) {
```

Comment:
@iefode should `generate_inputs()` be moved to `inputsMap`?

Reply:
@vurusovs Of course! Thanks!
@olpipi Could you please create a new PPR to apply comment?

Reply:
Yes, I can try it.
But it can affect other tests. So need to check CI